rfcs/003-use-black.md

```diff
@@ -0,0 +1,96 @@
+
+## Reference-level Explanation (aka Technical Details)
+
+[This PR](https://github.com/amundsen-io/amundsendatabuilder/pull/405) implements the change for the Amundsen-Databuilder codebase. It:
+
+- makes a few changes to unnest some deeply nested data structures
+- applies `black` to the entire codebase
+- configures `black` to use 120 lines (which was how `flake8` had been configured)
+- configures `flake8` to be compatible with Black
+
+We may also want to figure out how to prevent a huge change to `git blame`. I am ambivalent on this. My argument against is that if anyone is using git blame and finds a line which was reformatted, they
+can just look at the previous blame. However, there are [solutions](https://github.com/psf/black#migrating-your-code-style-without-ruining-git-blame).
+
+## Drawbacks
+
+- Updating `blame` for hundreds of lines would be minor drawback, but as mentioned, that can probably be resolved.
+- The new style is a significant deviation from the existing style.
+    - Some apsects are trivial, e.g. quotes are swapped to `"`,
+    - I argue that most of the changes are improvements,
+    - ... but Black does have trouble with very long identifiers/strings and deeply nested data structures, and Amundsen includes a fair number of these
+      due to the nature of its configuration.
+        - ... but in my opinion, having decided that the codebase is Python 3.6+ only, I think using `f"strings"` could help alleviate those areas.
+        - ... and also, you can factor highly nested structures, declare long strings as module level constants, and stuff like that, which I think help with code quality anyway.
+
+## Alternatives
+
+- The main alternative is YAPF. If the majority here really hate Black's decisions, YAPF still solves the core issue.
+- If we like Black except for `"double quotes"`... there is an alternative called [Brunette](https://github.com/odwyersoftware/brunette) which includes a `single-quotes` option.
+- Other possibilities are less strict formatters e.g. autopep8. If people desparately still want some formatting flexibility, then this would be a compromise which is better than nothing.
+
+Ultimately, the stakes here aren't actuall _that_ high. If we do not do this, we will just be stuck with the problems I described in the Motivation section.
+
+## Prior art
+
+Black has a [used by](https://github.com/psf/black#used-by) section (which could probably be much bigger!) which includes Facebook, Dropbox, Pytest. Brex (my employer) also uses it successfully.
+Google uses YAPF.
+
+## Unresolved questions
+
+- Should we include a pre-commit hook?
+- Will we resolve the `git blame` issue? And how?
+- If we choose Black, should we go with a line-length of 120?
+    - I chose that in my PR because flake8 had been configured that way
+    - Since the Databuilder repo seems prone to some particularly long lines, I think 120 probably makes sense; but I'm ambivalent.
+- Which repos will this apply to?
```

Comment:
in additiojn to databuilder, I i think we should apply to amundsenfrontendservice, amundsenmetadataservice, amundsensearchservice and common -- as mentioned, I'm happy to do the implementation, but I think we should standardize across all of the repos

Reply:
yeah, if we decide to enforce linter, we should apply to all the other repos as well.

Reply:
The more the merrier! I've only been working closely with Databuilder, which is the reason that's where my focus has been